tests/api/test_custom_index.py

```diff
@@ -167,7 +167,8 @@ def annotate_feed(self, feed, lane):
         def mock_url_for(controller, library_short_name, **kwargs):
             """Create a real-looking URL for any random controller."""
             url_for_calls.append((controller, library_short_name, kwargs))
-            query = "&".join([f"{k}={v}" for k, v in sorted(kwargs.items())])
+            filtered_args = {k: v for k, v in kwargs.items() if not k.startswith("_")}
```

Comment:
This `mock_url_for` function is only used in the `test__navigation_feed` test function. It takes any `kwargs` passed into the function, and turns them into query parameters, similar to the normal flask `url_for` function. 

Flasks `url_for` method takes a number of parameters prefixed with `_` that it does not include in the URL. This updates the mock to reflect that behavior. 
https://flask.palletsprojects.com/en/1.1.x/api/#flask.url_for

We could be more specific and look for `_external`, but it seemed to me like it was better to strip all the `kwargs` that start with `_` in case we decide to use one of the other possible parameters in the future, so we don't break the test mock again in that case.

Reply:
Probably worth adding a comment to this effect here.

Reply:
👍🏻 added